plugin/path_role.go

```diff
@@ -12,71 +13,133 @@ import (
 )
 
 const (
-	iamRoleType            = "iam"
-	serviceAccountAttr     = "service-accounts"
-	serviceAccountWildcard = "*"
+	// Role types
+	iamRoleType = "iam"
+	gceRoleType = "gce"
 
+	// Errors
 	errEmptyRoleName           = "role name is required"
+	errEmptyRoleType           = "role type cannot be empty"
+	errEmptyProjectId          = "project id cannot be empty"
 	errEmptyIamServiceAccounts = "IAM role type must have at least one service account"
+
+	errTemplateEditListWrongType   = "role is type '%s', cannot edit attribute '%s' (expected role type: '%s')"
+	errTemplateInvalidRoleTypeArgs = "invalid args found for role of type %s: %s"
+
+	// Other
+	serviceAccountsWildcard = "*"
+
+	// Default duration that JWT tokens must expire within to be accepted (currently only IAM)
+	defaultIamMaxJwtExpMinutes int = 15
+
+	// Max allowed duration that all JWT tokens must expire within to be accepted
+	maxJwtExpMaxMinutes int = 60
 )
 
+var baseRoleFieldSchema map[string]*framework.FieldSchema = map[string]*framework.FieldSchema{
+	"name": {
+		Type:        framework.TypeString,
+		Description: "Name of the role.",
+	},
+	"type": {
+		Type:        framework.TypeString,
+		Description: "Type of the role. Currently supported: iam",
+	},
+	"policies": {
```

Comment:
This should be a TypeCommaStringSlice, which lets people specify it as either a "comma,separated,string" or via JSON as a ["json", "array"] !

Reply:
done